panimg/image_builders/dicom.py

```diff
@@ -264,7 +264,7 @@ def _create_itk_from_dcm(self, z_order: int) -> SimpleITK.Image:
             self.ref_header, "PhotometricInterpretation", None
         )
         if not is_rgb and photometric_interpretation == "MONOCHROME1":
-            dcm_array -= np.max(dcm_array)
+            dcm_array = np.max(dcm_array) - dcm_array
```

Comment:
What will happen if the array is of a signed type? Say int16 and has both -32,768 and +32,768. Wouldn't this overflow the data type?

Reply:
Maybe using `np.invert()` would be the better call here? But I am no expert in these kinds of data manipulation and shifts.

Reply:
Yes, you're right - `np.invert()` would indeed fix that but doesn't work for float arrays, but we can just multiply those with -1. There might be neater ways to do this where arrays with only positive numbers but a signed data type have still only positive numbers afterwards, or where unsigned arrays with small values still have only small values afterwards.